draft/ZEP0010.md

```diff
@@ -0,0 +1,413 @@
+        "reference": "../.."
+      }
+    }
+  ]
+
+}
+```
+
+4. A shared or even core reference extension
+
+```javascript
+{
+  "extensions":  [
+    {
+      "name": "example.parent-ref",
+      "configuration": {
+        "reference": "../.."
+      }
+    }
+  ]
+}
+```
+
+As further experience is gained by the community of extension authors,
+one or more of these methods may be adopted into the core spec.
+
+### Alternatives for the `extensions` extension point
+
+The current design allows having the same
+extension definition syntax across all extension points and reduces pollution
+of the top-level namespace in a `zarr.json`. Thus, the addition of top-level
+metadata keys remains reserved to changes in the core spec. This MAY happen as
+part of the core spec adopting functionality of an extension.
+
+Alternative designs that were considered are listed below along with their
+pros and cons.
+
+#### Top-level metadata keys
+
+Instead of a generic extension point, new top-level
+extension keys could be added to the metadata::
+
+```javascript
+{
+    "zarr_format": 3,
+    ...
+    "example.offset": { "offset": [ 12 ] },
+    "example.array-statistics": {
+        "min": 5,
+        "max": 12
+    },
+    "example.consolidated-metadata": {
+        "must_understand": false,
+        ...
+    }, // optional extension
+    ...
+}
+```
+
+In this case, there would be no explicit `configuration` key within an
+extension definition, but instead all the keys of such a configuration would be
+in the object itself. Using an object rather than directly for example
+an array of values would allow for evolution of the extension.
+
+This would mean, however, that there are two separate types of
+extension definitions, i.e. `{"name":"<name>", "configuration": {...}}` in
+specialized extension points (e.g. `codecs`) and `"<name>": {...}` for other
+extensions.
+
+A benefit would be that if an extension becomes adopted into the core spec, implementations
+would not need to be updated to support their move from the ``extensions`` object.
+
+#### Simple `extensions` object
+
+Instead of an array that holds the extension definitions, an object could alternatively be used::
+
+```javascript
+{
+    "zarr_format": 3,
+    ...
+    "extensions": {
+        "example.offset": { "offset": [ 12 ] },
+        "example.array-statistics": {
+            "min": 5,
+            "max": 12
+        },
+        "example.consolidated-metadata": {
+            "must_understand": false,
+            ...
+        } // optional extension
+    },
+    ...
+}
+```
+
+This alternative is similar to the top-level keys, with mostly the same implications.
+
+This alternative would continue to reserve the top-level namespace for changes to
+the core spec and, therefore, reduce pollution of the top-level namespace. Downsides include
+that only a single use of each extension would be possible since the key is the extension
+name and there would be no ordering of the extensions.
```

Comment:
Since lack of an ordering is listed as a disadvantage, what do you see as a use case for ordering of extensions or listing the same extension more than once?